core/kernel/persistence/starsql/class.Resource.php

```diff
@@ -543,22 +543,28 @@ public function getPropertiesValues(core_kernel_classes_Resource $resource, $pro
         foreach ($result->get('resource')->getProperties() as $key => $value) {
             if (in_array($key, $propertyUris)) {
                 if (is_iterable($value)) {
-                    $returnValue[$key] = array_merge(
-                        $returnValue[$key] ?? [],
-                        $this->getLanguageProcessor()->filterByLanguage($value, [$dataLanguage, $defaultLanguage])
-                    );
+                    $returnValue[$key] =
```

Comment:
**nitpick**: could you consider refactoring this part (extracting method refactoring in particular) as it becomes a bit difficult to read?